frontend/src/components/dashboard/CodeComparison.tsx

```diff
@@ -589,16 +634,21 @@ const CodeComparison = ({
                   <Button
                     onClick={() => {
                       if (isRightSolutionBookmarked) {
-                        setBookmarkedSolutionIds(
-                          bookmarkedSolutionIds.filter(
-                            (id) => id !== rightDataPoint.analysis.solutionId,
-                          ),
-                        );
+                        dispatch({
+                          type: AnalyzerStateActionType.setBookmarkedSolutions,
+                          bookmarkedSolutionIds:
+                            state.bookmarkedSolutionIds.filter(
+                              (id) => id !== rightDataPoint.analysis.solutionId,
+                            ),
```

Comment:
I'm wondering if here we should do an `addBookmarkedSolution` / `removeBookmarkedSolution` instead, moving the logic to the reducer as well